src/transformers/models/sam/image_processing_sam.py

```diff
@@ -392,11 +396,406 @@ def post_process_masks(
         for i, original_size in enumerate(original_sizes):
             interpolated_mask = F.interpolate(masks[i], target_image_size, mode="bilinear", align_corners=False)
             interpolated_mask = interpolated_mask[..., : reshaped_input_sizes[i][0], : reshaped_input_sizes[i][1]]
-            interpolated_mask = F.interpolate(
-                interpolated_mask, [*original_size.numpy()], mode="bilinear", align_corners=False
-            )
+            interpolated_mask = F.interpolate(interpolated_mask, original_size, mode="bilinear", align_corners=False)
             if binarize:
                 interpolated_mask = interpolated_mask > mask_threshold
             output_masks.append(interpolated_mask)
 
         return output_masks
+
+    def post_process_for_mask_generation(self, all_masks, all_scores, all_boxes, crops_nms_thresh):
+        """
+        Post processes mask that are automatically generated.
+        """
+        return _postprocess_for_amg(all_masks, all_scores, all_boxes, crops_nms_thresh)
```

Comment:
nit: reflect naming update (remote 'automatic) and make docstring clearer 

```suggestion
        Post processes generated masks
        """
        return _postprocess_for_mg(all_masks, all_scores, all_boxes, crops_nms_thresh)
```